### PR TITLE
Feature/remove unused export

### PR DESCRIPTION
**Pull request by McDaddy**
## What this PR does / why we need it:
1. convert all router export to default export
2. remove `import * as xxx` syntax
3. remove all unused export
4. if something is used only in the local component scope, then remove `export`

I just finished part of the job. Rest of them might need help from Huiluo to complete .

image

## Does this PR introduce a user interface change?

❎ No

## ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |              |
| 🇨🇳 中文    |              |

## Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: shell/tools/find-unused-module.ts [hunk 1]
```diff
@@ -38,16 +38,27 @@ function findUnusedModule(options: Options) {
...
   entries.forEach((entry) => {
     const entryPath = resolve(cwd, entry);
     entryModules.push(entryPath);
-    traverseModule(entryPath, (modulePath: string) => {
+    traverseModule(entryPath, (modulePath: string, items: string[], isImport: boolean) => {
       usedModules.push(modulePath);
+      if (isImport) {
+        importModules = importModules.concat(...items);
+      } else {
+        exportModules = exportModules.concat(...items);
```
Comment (daskyrk) [inline shell/tools/find-unused-module.ts:53]: why not use push?

File: shell/tools/find-unused-module.ts [hunk 2]
```diff
@@ -38,16 +38,27 @@ function findUnusedModule(options: Options) {
...
+      }
     });
   });
 
+  const importSet = new Set(importModules);
+  const exportSet = new Set(exportModules);
+  const unusedExport = new Set([...exportSet].filter((x) => !importSet.has(x)));
```
Comment (daskyrk) [inline shell/tools/find-unused-module.ts:60]: This set is useless ? ExportSet is already an unduplicated set, which means its subset is also an unduplicated array.

File: shell/tools/detect-unused-files.ts [hunk 3]
```diff
@@ -65,6 +65,8 @@ const detect = () => {
   (result.unused || []).forEach((item) => {
     console.log(item);
   });
+  console.log(`unused export count: ${result.unusedExport.size}`);
+  console.log(result.unusedExport);
```
Comment (daskyrk) [inline shell/tools/detect-unused-files.ts:69]: This script only do log but not modify original file directly, right ?

Comment (McDaddy) [reply shell/tools/detect-unused-files.ts:69]: yep, just remind people which ones are unused, delete operation should be done manually

**Review by daskyrk — APPROVED** [review]

**Pull request merged**